includes/class-wc-payments.php

```diff
@@ -1301,4 +1323,28 @@ public static function enqueue_dev_runtime_scripts() {
 			wp_enqueue_script( 'WCPAY_RUNTIME', plugins_url( 'dist/runtime.js', WCPAY_PLUGIN_FILE ), [], self::get_file_version( 'dist/runtime.js' ), true );
 		}
 	}
+
+	/**
+	 * Creates a new request object for a server call.
+	 *
+	 * @param  string $class_name The name of the request class. Must extend WCPay\Core\Server\Request.
+	 * @return Request
+	 * @throws Exception          If the request class is not really a request.
+	 */
+	public static function create_request( $class_name ) {
```

Comment:
Why do we need the factory method here? Is there a strong reason not to put it in the `Request` class? 

We're losing valuable typehints in IDE with this move.

Reply:
> Why do we need the factory method here?

Dependencies. I think it was added in the context of level5 data, but overall this allows the request to be created with some dependencies, which are otherwise private to the `WC_Payments` class.

On the other hand, I would not mind the request constructor/static method to do call something like `WC_Payments::inject_server_request_dependencies( $this )` in there. This way we could still only provide those dependencies to the request without making them truly public.

Reply:
Couldn't we use `wc_get_container()` for this?